toolkit/tools/bldtracker/bldtracker.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to track build time during image generation.
+
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app          = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
+	scriptName   = app.Flag("script-name", "The name of the current tool.").Required().String()
+	stepName     = app.Flag("step-name", "The name of the current step.").Required().String()
+	actionName   = app.Flag("action-name", "The name of the current sub-action.").Default("").String()
+	filePath     = app.Flag("file-path", "The folder that stores timestamp csvs.").Required().String()                          // currently must be absolute
```

Comment:
You can handle directory existence validation here using [`.ExistingDir()`](https://pkg.go.dev/gopkg.in/alecthomas/kingpin.v2#FlagClause.ExistingDir).

```suggestion
	filePath     = app.Flag("file-path", "The folder that stores timestamp csvs.").Required().ExistingDir()                          // currently must be absolute
```